enterprise/com/src/main/java/org/neo4j/com/storecopy/TransactionCommittingResponseUnpacker.java

```diff
@@ -191,7 +201,86 @@ public void unpackResponse( Response<?> response, final TxHandler txHandler ) th
         {
             if ( response.hasTransactionsToBeApplied() )
             {
-                applyQueuedTransactions();
+                applyQueuedTransactionsIfNeeded();
+            }
+        }
+    }
+
+    private void applyQueuedTransactionsIfNeeded() throws IOException
+    {
+        if ( transactionQueue.isEmpty() )
+        {
+            return;
+        }
+
+        /**
+         * Problem 1 (Not really a problem but we thought it where)
+         *  - chunk of batch is smaller than safe zone
+         *  - tx started after activeTransactions() is called
+         *  is safe because those transactions will see the latest state of store before chunk is applied and
+         *  because chunk is smaller than safe zone we are guarantied to not see two different states of any record
+         *  when applying the chunk.
+         *
+         *    activeTransactions() is called
+         *    |        start committing chunk
+         * ---|----+---|--|------> TIME
+         *         |      |
+         *         |      Start applying chunk
+         *         New tx starts here. Does not get terminated because not among active transactions, this is safe.
+         *
+         * Problem 2
+         *  - chunk of batch is larger than safe zone
+         *  - tx started after activeTransactions() but before apply
+         *
+         *    activeTransactions() is called
+         *    |        start committing chunk
+         * ---|--------|+-|------> TIME
+         *              | |
+         *              | Start applying chunk
+         *              New tx starts here. Does not get terminated because not among active transactions, but will read
+         *              outdated data and can be affected by reuse contamination.
```

Comment:
I'd like to see the two different parts of problem two explained separately. The diagram only explains `tx started after activeTransactions() but before apply`.